reconnecting-websocket.ts

```diff
@@ -422,13 +414,29 @@ export default class ReconnectingWebSocket {
         }
         this._removeListeners();
         try {
-            this._ws.close(code, reason);
+            this._safeClose(code, reason);
             this._handleClose(new Events.CloseEvent(code, reason, this));
         } catch (error) {
             // ignore
         }
     }
 
+    private _safeClose(code: number, reason?: string) {
+        if (!this._ws) {
+            this._debug('close enqueued: no ws instance');
+            return;
+        }
+        if (this._ws.readyState === this.CLOSING) {
+            this._debug('close enqueued: already closing');
+            return;
+        }
+        if (this._ws.readyState === this.CLOSED) {
+            this._debug('close: already closed');
+            return;
+        }
+        this._ws.close(code, reason);
```

Comment:
Seems a little confusing to me, I think we actually did need to close the `CONNECTING` connection.

I think the error you say maybe happened on we close a `CLOSING` socket instead of a `CONNECTING` socket.

we see the `ws` code, when we close the `CONNECTING` connection, the `ws` package will change its state to `CLOSING`, emit `error` and `close` event in the next tick.
<img width="741" alt="image" src="https://github.com/opensumi/reconnecting-websocket/assets/13938334/0b98d597-f15f-4691-864d-1fc4415d1ca3">

for code in the `master` branch, when it receive `error` event, it will invoke `_disconnect` to close the original ws connection, but here the connection is already CLOSING or CLOSED, it may raise an error.

Reply:
And I cannot understand why the **process will exit**, I will try to reproduce it.